crypto/evp/evp_enc.c

```diff
@@ -1393,6 +1408,30 @@ EVP_CIPHER *EVP_CIPHER_fetch(OPENSSL_CTX *ctx, const char *algorithm,
     return cipher;
 }
 
+int EVP_CIPHER_up_ref(EVP_CIPHER *cipher)
+{
+    int ref = 0;
+
+    CRYPTO_UP_REF(&cipher->refcnt, &ref, cipher->lock);
```

Comment:
Why do some .c files do UP_REF() <= 0 check?

Reply:
Good question.  I can give reasons why not to here: 1) we know that a non-const cipher has a correctly initialized lock, or it wouldn't exist, 2) we know that the locks are initialized to work "normally", i.e. corresponding to the "fast" mode in Linux pthread terms.

Reply:
I did find a fault in the corresponding function in `crypto/evp/digest.c`, though, thanks to this remark, so thanks for the indirect heads up :-)